core/common/src/main/java/alluxio/worker/block/io/LocalFileSecureBlockReader.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block.io;
+
+import alluxio.metrics.MetricKey;
+import alluxio.metrics.MetricsSystem;
+import com.codahale.metrics.Counter;
+import com.google.common.base.Preconditions;
+import com.google.common.io.Closer;
+import io.netty.buffer.ByteBuf;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.FileDescriptor;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ReadableByteChannel;
+
+/**
+ * This class provides read access to a block data file locally stored in managed storage.
+ */
+@NotThreadSafe
+public class LocalFileSecureBlockReader extends BlockReader {
```

Comment:
is it possible to reuse the code in LocalFileDataReader?  make LocalFileSecureDataReader inherit LocalFileDataReader?